sale_commission/wizard/wizard_settle.py

```diff
@@ -18,95 +19,90 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from openerp import models, fields
 
+from openerp import models, fields, api, exceptions, _
+from datetime import date, timedelta
+from dateutil.relativedelta import relativedelta
 
-class settled_wizard (models.TransientModel):
-    """settled.wizard"""
 
-    _name = "settled.wizard"
+class SaleCommissionMakeSettle(models.TransientModel):
+    _name = "sale.commission.make.settle"
 
-    date_from = fields.Date('From', required=True)
-    date_to = fields.Date('To', required=True)
+    date_to = fields.Date('Up to', required=True, default=fields.Date.today())
+    agents = fields.Many2many(comodel_name='res.partner',
+                              domain="[('agent', '=', True)]")
 
-    def settlement_exec(self, cr, uid, ids, context=None):
-        """Execute correctly for both."""
-        if context is None:
-            context = {}
-        pool_liq = self.pool['settlement']
-        for o in self.browse(cr, uid, ids, context=context):
-            vals = {
-                'name': o.date_from + " // " + o.date_to,
-                'date_from': o.date_from,
-                'date_to': o.date_to
-            }
-            liq_id = pool_liq.create(cr, uid, vals, context=context)
-            pool_liq.calculate(
-                cr, uid, liq_id,
-                context['active_ids'],
-                o.date_from,
-                o.date_to,
-                context=context
-            )
+    def _get_period_start(self, agent, date_to):
+        if isinstance(date_to, basestring):
+            date_to = fields.Date.from_string(date_to)
+        if agent.settlement == 'monthly':
+            return date(month=date_to.month, year=date_to.year, day=1)
+        elif agent.settlement == 'quaterly':
+            # Get first month of the date quarter
```

Comment:
You're totally right: I have to take the beggining of this quarter, not the next one.